tests/Core/Tokenizers/Tokenizer/RecurseScopeMapSwitchTokenScopeTest.inc

```diff
@@ -22,3 +22,12 @@ switch ($value):
         break;
 /* testSwitchAlternativeSyntaxEnd */
 endswitch;
+
+// Test for https://github.com/squizlabs/PHP_CodeSniffer/issues/543
+/* testSwitchClosureWithinCondition */
+switch((function () {
```

Comment:
The test could be made even more "interesting" if the closure had a return type declaration (which also uses the `T_COLON` token).

Might actually even be better to have a second test sample for that.

On that note, maybe we should also have a test with an arrow function callback with/without return type declaration in a switch condition ?

**Note:** the above remarks do not need to be addressed in this PR. They can be left for a follow-up PR to further improve the tokenizer tests.